Source/Urho3D/IO/Serializer.h

```diff
@@ -140,4 +140,7 @@ class URHO3D_API Serializer
     bool WriteLine(const ea::string& value);
 };
 
-}
+/// Write a string data without zero termination. This should be last or the only write into serializer.
+bool WriteStringData(Serializer* serializer, const ea::string& value);
```

Comment:
Hmm now that i think about it i think we should not have these at all. This is essentially `Serializer::Write()`/`Deserializer::Read()` with a string type. If you really insist on reading/writing arbitrary data them imo use `Read()`/`Write()`. Yes, it is not convenient, but that is the point. Such data reading/writing should be avoided, therefore it is good that it is not convenient. If you need a `Resource`-backed plain text file then right approach would be to look at `JSONFile` and create a `BinaryFile` and `TextFile` that inherits `BinaryFile`. `TextFile` would pretty much just append a null terminator and provide a way to get a string.

Reply:
A reminder - I need this to use from c#. To use Write I would need to manually create an array from string, get a pointer and execute Read, then get rid of array. To use Read I need to allocate array of necessary size, run Read into it, create a string from it and then get rid of array.

"and TextFile that inherits BinaryFile" - how this new file class will help me to read or write resource if the resource interface takes Serializer and Deserializer as an argument, not a particular file type?